cmd/state-installer/cmd.go

```diff
@@ -260,17 +261,10 @@ func execute(out output.Outputer, cfg *config.Instance, an analytics.Dispatcher,
 		params.isUpdate = determineLegacyUpdate(stateToolInstalled, packagedStateExe, payloadPath, params)
 	}
 
-	// If the state tool is already installed, but out of date, try to update it first.
-	if stateToolInstalled && !params.isUpdate && !params.force {
-		checker := updater.NewDefaultChecker(cfg)
-		up, err := checker.CheckFor("", "")
-		if err != nil {
-			logging.Debug("Could not check for update.") // not a show-stopper; move on
-		}
-		if up != nil {
-			logging.Debug("Update available. Switching to update flow.")
-			params.isUpdate = true
-		}
+	// If the state tool is already installed, but out of date, continue with update flow.
+	if stateToolInstalled && !params.isUpdate && !params.force && shouldUpdateInstalledStateTool() {
```

Comment:
```suggestion
	if stateToolInstalled && ((!params.isUpdate && !params.force) || shouldUpdateInstalledStateTool()) {
```

Note also filed a story to clean this up:

https://activestatef.atlassian.net/browse/DX-1188

Reply:
I don't like this because I think we should only call out to `state --version` when absolutely necessary (not already in an update flow, and user is not explicitly forcing install). With your proposed solution, I think we're going to call out if we're already in an update flow OR the user is forcing the install.